src/NuGet.Services.AzureSearch/SearchService/AzureSearchQueryBuilder.cs

```diff
@@ -38,171 +38,175 @@ internal class AzureSearchQueryBuilder
             '+', '-', '&', '|', '!', '(', ')', '{', '}', '[', ']', '^', '"', '~', '*', '?', ':', '\\', '/',
         };
 
-        private readonly List<string> _nonFieldScopedValues;
-        private readonly Dictionary<string, List<string>> _fieldScopedValues;
+        private readonly StringBuilder _result;
+        private int _clauses;
 
         public AzureSearchQueryBuilder()
         {
-            _nonFieldScopedValues = new List<string>();
-            _fieldScopedValues = new Dictionary<string, List<string>>();
+            _result = new StringBuilder();
+            _clauses = 0;
         }
 
-        public void AddNonFieldScopedValues(IEnumerable<string> values)
+        public void AddTerms(IReadOnlyList<string> terms)
```

Comment:
I think adding comments for the methods will be valuable.  Is there a specific AzureSearchQuery syntax that we try to follow? If yes, a link to docs would be useful . 

Is it allowed to call AddTerms and AddScopedTerms on the same instance or invoke multiple times the same method?

Reply:
I've improved the documentation in this class. Please let me know if you have any additional suggestions!